python/pyspark/sql/catalog.py

```diff
@@ -138,6 +139,78 @@ def listColumns(self, tableName, dbName=None):
                 isBucket=jcolumn.isBucket()))
         return columns
 
+    @ignore_unicode_prefix
+    @since(3.0)
+    def databaseExists(self, dbName):
+        """Check if the database with the specified name exists."""
+        return self._jcatalog.databaseExists(dbName)
+
+    @ignore_unicode_prefix
+    @since(3.0)
+    def functionExists(self, functionName, dbName=None):
```

Comment:
Hm, this signature looks different with Scala side:

```scala
  def functionExists(functionName: String): Boolean
  def functionExists(dbName: String, functionName: String): Boolean
```

Can we handle this? There're some examples to refer, for instance,

https://github.com/apache/spark/blob/ecaa495b1fe532c36e952ccac42f4715809476af/python/pyspark/sql/dataframe.py#L811-L846

Reply:
I think as written is more Python-standard API. Certainly we can do interesting things with vargs/kwargs to make this look more like Scala, but I don't think that would be ideal here. One possibility could be requiring only kwargs (named arguments) so it's not ambigious. What do you think?